pdns/dnsdist-lua.cc

```diff
@@ -110,11 +110,15 @@ std::shared_ptr<DNSRule> makeRule(const luadnsrule_t& var)
   };
   if(auto src = boost::get<string>(&var))
     add(*src);
-  else {
-    for(auto& a : boost::get<vector<pair<int, string>>>(var)) {
+  else if (auto src = boost::get<vector<pair<int, string>>>(&var))
```

Comment:
g++ with `-Wshadow` is going to complain we declare several `src` in the same scope, perhaps we could avoid that?